metals/src/main/scala/scala/meta/internal/metals/codeactions/ExtractValueCodeAction.scala

```diff
@@ -33,55 +33,131 @@ class ExtractValueCodeAction(
+    }
+    val textEdits: Option[Seq[(Seq[l.TextEdit], String)]] = {
+      for {
+        head <- allTrees.headOption
+        stats <- lastEnclosingStatsList(head)
+        name = createNewName(stats)
+        stat <- stats.find(stat => stat.pos.encloses(head.pos))
+        source <- buffers.get(path)
+      } yield {
+        val blank =
+          if (source(stat.pos.start - stat.pos.startColumn) == '\t') '\t'
+          else ' '
+        val indent = blank.stringRepeat(indentationLength(source, stat.pos))
+        val keyword = if (stat.isInstanceOf[Enumerator]) "" else "val "
+        // we will insert `val newValue = ???` before the existing statement containing apply
+        for {
+          term <- allTrees
+          argument <- findRangeEnclosing(term, range)
+          // avoid extracting lambdas (this needs actual type information)
+          if isNotLambda(argument)
+        } yield {
+          val valueText = s"${indent}$keyword$name = ${argument.toString()}"
+          val valueTextWithBraces = withBraces(stat, source, valueText, blank)
+          // we need to add additional () in case of  `apply{}`
+          val replacementText =
+            term match {
+              case apply: Term.Apply
+                  if argument.is[Term.Block] && !applyHasParens(apply) =>
+                s"($name)"
+              case _ => name
+            }
+          val replacedArgument =
+            new l.TextEdit(argument.pos.toLSP, replacementText)
+          (valueTextWithBraces :+ replacedArgument, argument.toString())
+        }
       }
-      // avoid extracting lambdas (this needs actual type information)
-      if isNotLambda(argument)
-      stat <- stats.find(stat => stat.pos.encloses(apply.pos))
-      name = createNewName(stats)
-      source <- buffers.get(path)
-    } yield {
-      val blank =
-        if (source(stat.pos.start - stat.pos.startColumn) == '\t') '\t' else ' '
-      val indent = blank.stringRepeat(indentationLength(source, stat.pos))
-      val keyword = if (stat.isInstanceOf[Enumerator]) "" else "val "
-      // we will insert `val newValue = ???` before the existing statement containing apply
-      val valueText = s"${indent}$keyword$name = ${argument.toString()}"
-      val valueTextWithBraces = withBraces(stat, source, valueText, blank)
-      // we need to add additional () in case of  `apply{}`
-      val replacementText =
-        if (argument.is[Term.Block] && !applyHasParens(apply)) s"($name)"
-        else name
-      val replacedArgument = new l.TextEdit(argument.pos.toLSP, replacementText)
-      valueTextWithBraces :+ replacedArgument
     }
+
     textEdits match {
-      case Some(edits) =>
-        val codeAction = new l.CodeAction()
-        codeAction.setTitle(ExtractValueCodeAction.title)
-        codeAction.setKind(this.kind)
-        codeAction.setEdit(
-          new l.WorkspaceEdit(
-            Map(path.toURI.toString -> edits.asJava).asJava
+      case Some(actions) =>
```

Comment:
I think `textEdits.getOrElse(Nil).map { case (edits, title) => ???}` will look better than match, it's always lesser amount of braces.

Reply:
Or we can add `.getOrElse(Nil)` directly to the `textEdits`, I mean sth like that
```scala
    val texEdits = {
    ...
    }.getOrElse(Nil)

    textEdits.map { case (edits, title) =>
```